src/main/java/org/webjars/WebJarExtractor.java

```diff
@@ -320,28 +325,14 @@ private String getFileNodeModuleIdEntry(File packageJsonFile) throws IOException
         return moduleId;
     }
 
-    private String getJsonNodeModuleId(String packageJson) {
-        String moduleId = null;
-        int namePosn = packageJson.indexOf(PACKAGE_JSON_NAME);
-        if (namePosn > -1) {
-            int moduleIdPosn = namePosn + PACKAGE_JSON_NAME.length();
-            while (moduleIdPosn < packageJson.length() && Character.isWhitespace(packageJson.charAt(moduleIdPosn)))
-                ++moduleIdPosn;
-            if (moduleIdPosn < packageJson.length() && packageJson.charAt(moduleIdPosn) == ':') {
-                ++moduleIdPosn;
-                while (moduleIdPosn < packageJson.length() && Character.isWhitespace(packageJson.charAt(moduleIdPosn)))
-                    ++moduleIdPosn;
-                if (moduleIdPosn < packageJson.length() && packageJson.charAt(moduleIdPosn) == '"') {
-                    ++moduleIdPosn;
-                    StringBuilder sb = new StringBuilder();
-                    while (moduleIdPosn < packageJson.length() && packageJson.charAt(moduleIdPosn) != '"') {
-                        sb.append(packageJson.charAt(moduleIdPosn++));
-                    }
-                    moduleId = sb.toString();
-                }
-            }
-        }
-        return moduleId;
+    String getJsonNodeModuleId(String packageJson) throws IOException {
+        ObjectMapper mapper = new ObjectMapper()
+                .configure(ALLOW_UNQUOTED_FIELD_NAMES, true)
+                .configure(ALLOW_SINGLE_QUOTES, true);
```

Comment:
Do you have any examples of where these are actually needed?  From [the docs](https://docs.npmjs.com/files/package.json):

> It must be actual JSON, not just a JavaScript object literal.